File: 807-custom-sort-string/custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        counts = Counter(s)
        res = []

        for c in order:
            if c in counts:
                res.append(c * counts[c])
                del counts[c]
        
        for char, count in counts.items():
            res.append(char * count)

        return "".join(res)

File: 807-custom-sort-string/custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        unordered_map<char, int> mp;
        for (char c : s) {
            mp[c]++;
        }

        string res = "";
        for (char c : order) {
            if (mp.find(c) != mp.end()) {
                res += string(mp[c], c);
                mp.erase(c);
            }   
        
        }

        for (auto& [c, count] : mp) {
            res += string(count, c);
        }

        return res;
    }
};